src/scripts/main.js

```diff
@@ -1,3 +1,21 @@
 'use strict';
 
-// write code here
+const list = document.querySelector('.tree');
+const listItems = document.querySelectorAll('li');
+
+for (const li of listItems) {
+  if (li.children.length > 0) {
+    const span = document.createElement('span');
+
+    li.prepend(span);
+    span.append(span.nextSibling);
+  }
+}
+
+list.addEventListener('click', (e) => {
+  const target = e.target;
+
+  if (target.tagName === 'SPAN') {
```

Comment:
It could be better to detect element by some another descriptor (like, custom data-attribute)